microsetta_public_api/models/_taxonomy.py

```diff
@@ -353,6 +353,51 @@ def get_group(self, ids: Iterable[str], name: str = None) -> GroupTaxonomy:
                              feature_variances=list(feature_variances),
                              )
 
+    def get_counts(self, level, samples=None) -> dict:
+        """Obtain the number of unique maximal specificity features
+
+        Parameters
+        ----------
+        level : int
+            The level to obtain feature counts for. A level of zero
+            indicates domain, one is phylum, etc.
+        samples : str or iterable of str, optional
+            The samples to collect data for. If not provided, counts are
+            derived from all samples.
+
+        Returns
+        -------
+        dict
+            The {taxon: count} observed. The count corresponds to the greatest
+            taxonomic specificity in the data. As an example, if counting at
+            the phylum level, and FOO had 2 classified genera and a classified
+            family without named genera, the count returned would be {FOO: 3}.
+        """
+        if samples is None:
+            table = self._table
+        elif isinstance(samples, str):
+            table = self._table.filter({samples, },
+                                       inplace=False).remove_empty()
+        else:
+            table = self._table.filter(set(samples),
+                                       inplace=False).remove_empty()
+
+        level_map = {0: 'Kingdom', 1: 'Phylum',
```

Comment:
My only concern with this approach is that it may not protect against taxonomy strings that skip levels or are malformed in some way.

E.g., Choosing to aggregate on Phylum on a table that includes a string like `k__Bacteria; o__Clostridiales`

Reply:
The `GreengenesFormatter` actually already helps safeguard against this, since it would properly map `o__Clostridiales` to `{'Order': 'Clostridiales'}`

Reply:
I think `level_map` should become a static attribute of `GreengenesFormatter` so this function is not hard-coded to that specific scheme.

Reply:
We do not support malformed taxonomies

Reply:
Okay then let's ignore the comments about malformed taxonomy strings and just move the `level_map` to `GreengenesFormatter` so the code is not here is not coupled to it and the map is more re-usable later.